### PR TITLE
Dataset, Source, Job Metrics

### PR DESCRIPTION
**Pull request by phixMe**
### Problem

https://github.com/user-attachments/assets/1894c875-822a-4b2d-9bd9-d05650a16ad0

This adds metrics for sources, datasets, and jobs.

- [ ] Add Migrations for `created_at` on the corresponding tables that are used with the number is known before a merge. 

### Solution

Various `generate_series` style queries to aggregate metrics cumulatively.

One-line summary: Three endpoints that return Intervals of counts for daily and weekly graphs on the new dashboard.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

**Comment by wslulciuc** [comment]
[PR #2906 - Chart for Dataset, Source, and Job counts](https://trello.com/c/sfI0ReDP/64-pr-2906-chart-for-dataset-source-and-job-counts)

File: api/src/test/java/marquez/db/StatsTest.java [hunk 1]
```diff
@@ -108,4 +118,93 @@ public void testGetStatsForLastDay() {
...
+        intervalMetricsDay.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countDay).isPresent();
+    assertThat(countDay.get()).isGreaterThanOrEqualTo(1);
+
+    List<IntervalMetric> intervalMetricsWeek = DB.lastWeekSourceMetrics("UTC");
+
+    Optional<Integer> countWeek =
+        intervalMetricsWeek.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countWeek).isPresent();
+    assertThat(countWeek.get()).isGreaterThanOrEqualTo(1);
```
Comment (wslulciuc) [inline api/src/test/java/marquez/db/StatsTest.java:208]: Is there a way to assert the weekly count we expect vs just ensuring it's `>=1`?

Comment (phixMe) [reply api/src/test/java/marquez/db/StatsTest.java:208]: The counts are attributed to different days depending on which time of day the test is run which is the primary challenge on this one.

Comment (phixMe) [reply api/src/test/java/marquez/db/StatsTest.java]: Ok, cleanup is in now.

**Review by wslulciuc — APPROVED** [review]
LGTM 💯 💯 🥇

**Pull request merged**